compiler/rustc_trait_selection/src/traits/select/mod.rs

```diff
@@ -1494,7 +1494,18 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
                             // will then correctly report an inference error, since the
                             // existence of multiple marker trait impls tells us nothing
                             // about which one should actually apply.
```

Comment:
Can this cause inference variables to be arbitrarily constrained, as described in the comments above?

e.g. if we had

```rust
#[marker]
pub trait F {}
impl F for u32 {}
impl<'a> F for &'a str where 'a: 'static {}
```

Would this change cause the `<_#0t as F>` obligation to unify `_#0t` with `u32`?

(contrived example because I don't know enough about trait selection/winnowing to provide an uncontrived one)

Reply:
no, because the `!needs_infer` condition still applies.

We drop strictly fewer candidates with this change.

Reply:
derp. boolean logic, it's no joke